src/chainstate/stacks/db/blocks.rs

```diff
@@ -3104,11 +3120,21 @@ impl StacksChainState {
         Ok((Some(epoch_receipt), None))
     }
 
+    /// Process staging blocks at the canonical chain tip,
+    ///  this only only needs to be used in contexts that aren't
+    ///  PoX aware (i.e., unit tests, and old stacks-node loops),
+    /// Elsewhere, block processing is invoked by the ChainsCoordinator,
+    ///  which handles tracking the chain tip itself
+    pub fn process_blocks_at_tip(&mut self, sort_db: &mut SortitionDB, max_blocks: usize) -> Result<Vec<(Option<StacksEpochReceipt>, Option<TransactionPayload>)>, Error> {
```

Comment:
Can you decorate this with `#[cfg(test)]` then?

Reply:
Addressed in https://github.com/blockstack/stacks-blockchain/pull/1787/commits/09a44c010ebea5b1e7be980529695a0fdd60b77f